tests/Unit/PointwiseFunctions/GeneralRelativity/Test_ComputeGhQuantities.cpp

```diff
@@ -186,22 +191,193 @@ void test_lapse_deriv_functions_analytic(
+      "TestFunctions", "spacetime_deriv_norm_shift",
+      {{{std::numeric_limits<double>::denorm_min(), 10.0}}}, used_for_size);
+}
+
+// Test computation of derivs of shift by comparing to Kerr-Schild
+template <typename Solution>
+void test_shift_deriv_functions_analytic(
+    const Solution& solution, const size_t grid_size_each_dimension,
+    const std::array<double, 3>& lower_bound,
+    const std::array<double, 3>& upper_bound) noexcept {
+  // Setup grid
+  const size_t SpatialDim = 3;
+  const size_t data_size = pow<SpatialDim>(grid_size_each_dimension);
+  Mesh<SpatialDim> mesh{grid_size_each_dimension, Spectral::Basis::Legendre,
+                        Spectral::Quadrature::GaussLobatto};
+
+  using Affine = CoordinateMaps::Affine;
+  using Affine3D = CoordinateMaps::ProductOf3Maps<Affine, Affine, Affine>;
+  const auto coord_map =
+      make_coordinate_map<Frame::Logical, Frame::Inertial>(Affine3D{
+          Affine{-1., 1., lower_bound[0], upper_bound[0]},
+          Affine{-1., 1., lower_bound[1], upper_bound[1]},
+          Affine{-1., 1., lower_bound[2], upper_bound[2]},
+      });
+
+  // Setup coordinates
+  const auto x_logical = logical_coordinates(mesh);
+  const auto x = coord_map(x_logical);
+  // Arbitrary time for time-independent solution.
+  const double t = std::numeric_limits<double>::signaling_NaN();
+
+  // Evaluate analytic solution
+  const auto vars =
+      solution.variables(x, t, typename Solution::template tags<DataVector>{});
+  const auto& lapse = get<gr::Tags::Lapse<>>(vars);
+  const auto& dt_lapse = get<Tags::dt<gr::Tags::Lapse<>>>(vars);
+  const auto& d_lapse =
+      get<typename Solution::template DerivLapse<DataVector>>(vars);
+  const auto& shift = get<gr::Tags::Shift<SpatialDim>>(vars);
+  const auto& d_shift_expected =
+      get<typename Solution::template DerivShift<DataVector>>(vars);
+  const auto& dt_shift_expected =
+      get<Tags::dt<gr::Tags::Shift<SpatialDim>>>(vars);
+  const auto& spatial_metric = get<gr::Tags::SpatialMetric<SpatialDim>>(vars);
+  const auto& dt_spatial_metric =
+      get<Tags::dt<gr::Tags::SpatialMetric<SpatialDim>>>(vars);
+  const auto& d_spatial_metric =
+      get<typename Solution::template DerivSpatialMetric<DataVector>>(vars);
+
+  // Get ingredients
+  const auto inverse_spatial_metric =
+      determinant_and_inverse(spatial_metric).second;
+  const auto inverse_spacetime_metric =
+      gr::inverse_spacetime_metric(lapse, shift, inverse_spatial_metric);
+  const auto phi =
+      GeneralizedHarmonic::phi(lapse, d_lapse, shift, d_shift_expected,
+                               spatial_metric, d_spatial_metric);
+  const auto pi =
+      GeneralizedHarmonic::pi(lapse, dt_lapse, shift, dt_shift_expected,
+                              spatial_metric, dt_spatial_metric, phi);
+  const auto normal_vector = gr::spacetime_normal_vector(lapse, shift);
+  const auto lower_shift = raise_or_lower_index(shift, spatial_metric);
+
+  // Compute dt_lower_shift
```

Comment:
We are also concerned that some of these tests (just the `dt_lower_shift` and `d4_norm_shift`, the others seem ok) are doing the same work as the functions they test (as in #1179). So, we'd like these to either be removed and rely solely on pypp tests, or if we can find some conveniently calculated mathematical identity, that could be checked instead. For these shift terms, I'm afraid I don't have a proposal for what could be calculated, but if you have an idea, that would be great :smile:.

Reply:
I have changed the test for `d4_norm_shift` to compute `d4_norm_shift = d4_spacetime_metric_00 + 2 lapse d4_lapse` instead and compare that to what is returned by its respective function. 

For `dt_lower_shift` I did not get a better idea than to test the identity `dt_norm_shift - lower_shift_dot_dt_shift = shift_dot_dt_lower_shift`. The test now (a) computes the LHS of this equality, (b) calls the function for `dt_lower_shift` (which we want to test) and takes its inner product with `shift`, and finally compares the result of (b) with (a). If you have a better idea, it would be more than welcome.

Reply:
I'm happy with these alterations to the tests.